builtin_map.go

```diff
@@ -68,9 +68,7 @@ func (mo *mapObject) export(ctx *objectExportCtx) interface{} {
 }
 
 func (mo *mapObject) exportToMap(dst reflect.Value, typ reflect.Type, ctx *objectExportCtx) error {
-	if dst.IsNil() {
-		dst.Set(reflect.MakeMap(typ))
-	}
+	dst.Set(reflect.MakeMap(typ))
```

Comment:
[q] curious why this nil check was removed. Is that not something we consider possible?

Reply:
I think the gist of it is that it doesn't matter anymore and they want to create a map when exporting no matter what (nil or not nil), they cover this a bit in the new test. Before it would only create it if the destination map was nil, now it doesn't matter. It will overwrite, if you look at `TestMapExportToNonNilMap` before this change that test would fail as the map `m` would not be overwritten, this this change it does. So if you were to add a check like:

```go
func TestMapExportToNonNilMap(t *testing.T) {
	vm := New()
	m := map[int]interface{}{
		2: true,
	}
	res, err := vm.RunString("new Map([[1, true]])")
	if err != nil {
		t.Fatal(err)
	}
	err = vm.ExportTo(res, &m)
	if err != nil {
		t.Fatal(err)
	}
	if len(m) != 1 {
		t.Fatal(m)
	}
        // THIS IS DIFFERENT TO UNDERSTAND
	// if _, exists := m[1]; !exists {
	//	 t.Fatal(m)
	// }
	if _, exists := m[2]; !exists {
		t.Fatal(m)
	}
}
```

It now fails but it would have passed before this change.